plaid/plaid.go

```diff
@@ -1,373 +1,147 @@
-}
+// LatestPlaidAPIVersion holds the latest version of the Plaid API
+const LatestPlaidAPIVersion = "2018-05-22"
 
+// Client holds information required to interact with the Plaid API.
 // Note: Client is only exported for method documentation purposes.
 // Instances should only be created through the 'NewClient' function.
 //
 // See https://github.com/golang/go/issues/7823.
 type Client struct {
-	clientID    string
-	secret      string
-	environment environmentURL
-	httpClient  *http.Client
-}
-
-type environmentURL string
-
-var Tartan environmentURL = "https://tartan.plaid.com"
-var Production environmentURL = "https://api.plaid.com"
-
-type Account struct {
-	ID      string `json:"_id"`
-	ItemID  string `json:"_item"`
-	UserID  string `json:"_user"`
-	Balance struct {
-		Available float64 `json:"available"`
-		Current   float64 `json:"current"`
-	} `json:"balance"`
-	Meta struct {
-		Number string `json:"number"`
-		Name   string `json:"name"`
-	} `json:"meta"`
-	Numbers struct {
-		Account     string `json:"account"`
-		Routing     string `json:"routing"`
-		WireRouting string `json:"wireRouting"`
-	} `json:"numbers"`
-	Type            string `json:"type"`
-	InstitutionType string `json:"institution_type"`
-}
-
-type Transaction struct {
-	ID        string `json:"_id"`
-	AccountID string `json:"_account"`
-
-	Amount float64 `json:"amount"`
-	Date   string  `json:"date"`
-	Name   string  `json:"name"`
-	Meta   struct {
-		AccountOwner string `json:"account_owner"`
-
-		Location struct {
-			Address string `json:"address"`
-			City    string `json:"city"`
-
-			Coordinates struct {
-				Lat float64 `json:"lat"`
-				Lon float64 `json:"lon"`
-			} `json:"coordinates"`
-
-			State string `json:"state"`
-			Zip   string `json:"zip"`
-		} `json:"location"`
-	} `json:"meta"`
-
-	Pending bool `json:"pending"`
-
-	Type struct {
-		Primary string `json:"primary"`
-	} `json:"type"`
-
-	Category   []string `json:"category"`
-	CategoryID string   `json:"category_id"`
-
-	Score struct {
-		Location struct {
-			Address float64 `json:"address"`
-			City    float64 `json:"city"`
-			State   float64 `json:"state"`
-			Zip     float64 `json:"zip"`
-		}
-		Name float64 `json:"name"`
-	} `json:"score"`
+	clientID        string
+	secret          string
+	publicKey       string
+	environment     Environment
+	httpClient      *http.Client
+	plaidAPIVersion string
 }
 
-type mfaIntermediate struct {
-	AccessToken string      `json:"access_token"`
-	MFA         interface{} `json:"mfa"`
-	Type        string      `json:"type"`
-}
-type mfaDevice struct {
-	Message string
-}
-type mfaList struct {
-	Mask string
-	Type string
-}
-type mfaQuestion struct {
-	Question string
-}
-type mfaSelection struct {
-	Answers  []string
-	Question string
+type ClientOptions struct {
+	ClientID        string
+	Secret          string
+	PublicKey       string
+	Environment     Environment
+	HTTPClient      *http.Client
+	PlaidAPIVersion string
```

Comment:
I think `APIVersion` would be more canonical.

Reply:
⚡️